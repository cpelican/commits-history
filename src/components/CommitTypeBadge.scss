
/* :root {
    --chore-color: darkgrey;
    --feature-color: orange;
    --build-color: green;
    --ci-color: purple;
    --style-color: violet;
    --refactor-color: aqua;
    --performance-color: chocolate;
    --test-color: firebrick;
} */

@mixin badge-variant($base-color) {
    color: rgba($base-color, 1);
    background-color: rgba($base-color, .1);
    border: 1px solid rgba($base-color, .5);
}
.commit-type-badge {
    display: inline;
    align-items: center;
    font-size: 12px;
    padding: 2px 4px;
    border-radius: 4px;
    font-weight: bolder;

    &.chore {
        @include badge-variant(darkgrey);
    }

    &.feature {
        @include badge-variant(orange);
    }

    &.build {
        @include badge-variant(green);
    }

    &.ci {
        @include badge-variant(purple);
    }

    &.style {
        @include badge-variant(violet);
    }

    &.refactor {
        @include badge-variant(aqua);
    }

    &.performance {
        @include badge-variant(chocolate);
    }

    &.test {
        @include badge-variant(firebrick);
    }
}