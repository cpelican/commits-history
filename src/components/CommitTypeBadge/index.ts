export {CommitTypeBadge, commitTypes} from './CommitTypeBadge';
