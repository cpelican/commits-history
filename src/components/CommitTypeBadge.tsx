import React from 'react';
import clsx from 'clsx';
import './CommitTypeBadge.scss';

export const commitTypes: Record<string, string> = {
    chore: 'chore',
    feature: 'feature',
    build: 'build',
    ci: 'ci',
    style: 'style',
    refactor: 'refactor',
    perf: 'performance',
    test: 'test'
};

export interface CommitTypeBadgeProps {
    type?: keyof typeof commitTypes;
}


export const CommitTypeBadge: React.FC<CommitTypeBadgeProps> = React.memo(({type}) => {
    if (type == null) {
        return null;
    }
    return (
        <span className={clsx('commit-type-badge', type)}>
            {type}
        </span>
    );
});

CommitTypeBadge.displayName = 'CommitTypeBadge';

