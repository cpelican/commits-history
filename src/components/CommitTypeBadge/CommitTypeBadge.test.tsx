import React from 'react';
import {render} from '@testing-library/react';
import {CommitTypeBadge, CommitTypeBadgeProps} from './CommitTypeBadge';

const defaultProps: CommitTypeBadgeProps = {
    type: 'ci',
};

describe('CommitTypeBadge', () => {
    test('renders as expected', () => {
        const {asFragment} = render(<CommitTypeBadge {...defaultProps} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders as expected with no type', () => {
        const {asFragment} = render(<CommitTypeBadge />);
        expect(asFragment()).toMatchInlineSnapshot('<DocumentFragment />');
    });
});
