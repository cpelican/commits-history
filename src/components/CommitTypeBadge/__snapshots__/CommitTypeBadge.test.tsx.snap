// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitTypeBadge renders as expected 1`] = `
<DocumentFragment>
  <span
    class="commit-type-badge ci"
  >
    ci
  </span>
</DocumentFragment>
`;
